Flutter/circular-charts/export-circular-chart.md

```diff
@@ -16,61 +16,201 @@ documentation: ug
-      final bytes = await data.toByteData(format: dart_ui.ImageByteFormat.png);
-      if (data != null) {
-        await Navigator.of(context).push(
-          MaterialPageRoute(
+        final ByteData? bytes =
+          await data.toByteData(format: ui.ImageByteFormat.png);
+        final Uint8List imageBytes =
+          bytes!.buffer.asUint8List(bytes.offsetInBytes, bytes.lengthInBytes);
+        await Navigator.of(context).push<dynamic>(
+          MaterialPageRoute<dynamic>(
             builder: (BuildContext context) {
-              return Scaffold(
-                appBar: AppBar(),
-                body: Center(
-                  child: Container(
-                    color: Colors.white,
-                    child: Image.memory(bytes!.buffer.asUint8List()),
-                  ),
-                ),
-              );
+              return Scaffold(body: Image.memory(imageBytes));
             },
           ),
         );
       }
     }
 
-{% endhighlight %}
-{% endtabs %}
+    class ChartSampleData {
+      ChartSampleData({this.x, this.y});
+        final String? x;
+        final num? y;
+    }
+
+  {% endhighlight %}
+  {% endtabs %}
 
 ## Export PDF
 
-Similar to the above way, we can also export the rendered chart as a PDF document. We create the pdf document using pdf component. This can be done in the application level itself and please find the code snippet below.
+Similar to the above way, we can also export the rendered chart as a PDF document. We create the pdf document using pdf component. This can be done in the application level itself and please find the code snippet below.It is necessary to include the platform-specific file generating codes to save the file as a PDF document. So, create two dart files ([`save_file_mobile`](https://github.com/syncfusion/flutter-examples/blob/master/lib/samples/pdf/helper/save_file_mobile.dart) and [`save_file_web`](https://github.com/syncfusion/flutter-examples/blob/master/lib/samples/pdf/helper/save_file_web.dart)) to save and launch the file in different platforms.
 
 {% tabs %}
 {% highlight dart %} 
 
-    Future<void> _renderCircularPDF() async {
-      var document = PdfDocument();
-      PdfPage page = document.pages.add();
-      dart_ui.Image data =
-          await _circularChartKey.currentState.toImage(pixelRatio: 3.0);
-      final bytes = await data.toByteData(format: dart_ui.ImageByteFormat.png);
-      final Uint8List imageBytes =
-          bytes!.buffer.asUint8List(bytes.offsetInBytes, bytes.lengthInBytes);
-      page.graphics
-          .drawImage(PdfBitmap(imageBytes), Rect.fromLTWH(25, 50, 300, 300));
-      var byteData = document.save();
-      document.dispose();
-      Directory directory = await getExternalStorageDirectory();
-      String path = directory.path;
-      File file = File('$path/Output.pdf');
-      await file.writeAsBytes(byteData, flush: true);
-      OpenFile.open('$path/Output.pdf');
+    import 'dart:typed_data';
```

Comment:
if we use this code, is the image and pdf exporting are working fine? We don't need to do any changes know?

Reply:
Yes. I have ensured with these codes by running in separate sample. It's working fine.